### ISSUE TITLE
Getting ads from AdGroupAd

### ISSUE THREAD

**Issue by keithslater** [Edujugon/laravel-google-ads#12]
Just doing something basic like this -

```
$data = $this->ads->service(AdGroupAdService::class)
    ->select('Id', 'Name')
            ->get();
        foreach ($data->items() as $d) {
            dd($d);
        }
```

For some reason it only returns 1 ExpandedAdText (I have multiple ads running. Even using getAd()). That Ad only returns an id & type, it doesn't return headline1, 2, urls, etc. Is there something I'm missing here? I'm trying to get my ad group ads along with the headline and url.

**Comment by Edujugon** [comment]
Hi @keithslater ,
In `select` method you must provide the exact list of fields you wanna retrieve from that service.

Give it a shot and let me know.

**Comment by keithslater** [comment]
How do I call fields that are part of another type. In the AdGroupAdService, there is an Ad class which has things like Url. How would I get Url out of there?

**Comment by Edujugon** [comment]
If you need just to fetch data,  [google reporting](https://github.com/Edujugon/laravel-google-ads#google-reports) may help you.

Give it a try and let me know.

**Comment by keithslater** [comment]
According to the docs, I shoudl be able to access the Url parameter however it doesn't return any Url. Same for Status and other fields.
```
$data = $this->ads->service(AdGroupAdService::class)
            ->select('Id', 'Name', 'Url', 'Headline', 'HeadlinePart1', 'SourceUrl', 'UrlData', 'Urls', 'DisplayUrl', 'Status')
            ->get();
        foreach ($data->items() as $d) {
            dd($d);
        }

```

https://developers.google.com/adwords/api/docs/appendix/selectorfields#v201710-AdGroupAdService

**Comment by Edujugon** [comment]
@keithslater , I'll have a look at that as soon as I get some free time.
I'll get back to you soon..

**Comment by Edujugon** [comment]
Hi @keithslater ,
Based on [google best practices](https://developers.google.com/adwords/api/docs/guides/bestpractices) you should use **reporting** to [get basic structural information](https://developers.google.com/adwords/api/docs/guides/zeroimpression-structure-reports#structure_reports). **services** are more for interacting with data but **reporting** are like the 'data reader'.

With this package you can easily get reports of any existing [report type](https://developers.google.com/adwords/api/docs/appendix/reports).

Just an example:
```php
$obj = $this->ads->report()
            ->from('CRITERIA_PERFORMANCE_REPORT')
            ->during('20170101','20170210')
            ->where('CampaignId = 752331963')
            ->select('CampaignId','AdGroupId','AdGroupName','Id', 'Criteria', 'CriteriaType','Impressions', 'Clicks', 'Cost', 'UrlCustomParameters')
            ->getAsObj();
```

As you can see, you can add conditions and more restrictions to the query..
For your specific case, The report type could be `AD_PERFORMANCE_REPORT`

You can check what fields are available for that specific report on google adwords doc BUT can also get them via code like follows:

```php
$fields = $this->ads->fields()->of('AD_PERFORMANCE_REPORT')->asList();
```
The above snippet will return an array of all fields available to that report. Easy, right?

So the next steps are even easier, just have a look at those fields, add them to the `select`method and done, you have your report ready.

Hope this helped :)

**Comment by keithslater** [comment]
Thanks for the info. 

I was able to get what I was trying to get by using CreativeFinalUrls in my original query.

I would use the ad reporting service however I don't want to have to parse a report in order to find the latest version of all the fields I'm looking for. Not to mention, I don't know the last time a change was made to the Ad and how far back I should be querying. I just want to get the latest version of an Ad which Google makes available in AdGroupAdService.

Either way I was able to get what I was looking for. Thanks.
